src/panels/lovelace/cards/hui-statistic-card.ts

```diff
@@ -184,6 +224,46 @@ export class HuiStatisticCard extends LitElement implements LovelaceCard {
     return true;
   }
 
+  protected willUpdate(changedProps: PropertyValues) {
+    super.willUpdate(changedProps);
+    if (!this._config || !changedProps.has("_config")) {
+      return;
+    }
+    const oldConfig = changedProps.get("_config") as
+      | StatisticCardConfig
+      | undefined;
+
+    if (this.hass) {
+      if (this._config.period === PERIOD_ENERGY && !this._energySub) {
+        this._subscribeEnergy();
+        return;
+      }
+      if (this._config.period !== PERIOD_ENERGY && this._energySub) {
+        this._unsubscribeEnergy();
+        this._setFetchStatisticTimer();
+        return;
+      }
+      if (
+        this._config.period === PERIOD_ENERGY &&
+        this._energySub &&
+        changedProps.has("_config") &&
+        oldConfig?.collection_key !== this._config.collection_key
+      ) {
+        this._unsubscribeEnergy();
+        this._subscribeEnergy();
+      }
+    }
+
+    if (
+      changedProps.has("_config") &&
+      oldConfig?.entity !== this._config.entity
+    ) {
+      this._fetchMetadata().then(() => {
+        this._setFetchStatisticTimer();
+      });
+    }
```

Comment:
This logic seems off. If you have `this._config.period === PERIOD_ENERGY` and change the entity, I think `this._setFetchStatisticTimer()` will be called

Reply:
I think that's ok/intended. 

`_setFetchStatisticTimer` fetches the statistic, and then schedules the timer only if period !== PERIOD_ENERGY. If it is using energy mode it will continue being subscribed to the energy callback. 

To be honest I'm not even sure if anything in this function even really matters... it looks like changing _anything_ in the config destroys the entire card element and rebuilds a new one from scratch. Like every time I touch the config I see `disconnectedCallback` and `firstUpdated` in hui-statistic-card being called again. I didn't think that was supposed to happen, seems odd.  Of course would still like to get it to be correct, but that surprised me.

Reply:
Recreating the card makes sense to me to avoid all this. But now I wonder why we bother with this update logic in all the cards. Perhaps custom things like `config-template-card` trigger this.
Anyway, approving this since it works.